src/rp2350/pico_platform/include/pico/asm_helper.S

```diff
@@ -64,24 +64,3 @@ weak_func WRAPPER_FUNC_NAME(\x)
 .macro __pre_init func, priority_string1
 __pre_init_with_offset func, 0, priority_string1
 .endm
-
```

Comment:
wonder if we should now include "hardware/hazard3.h" from asm_helper.S (and add the dependency to pico_platform on rp2350

Reply:
Actually, I'll split the instructions into a separate header (which is also included by the full `hazard3.h`)

Reply:
4269950 pulls in the Hazard3 instructions (only) in asm_helper. I broke up the hazard3 header a bit to support this.

Reply:
> I can do; doesn't stop me from pulling it in explicitly.

Yeah, my point was just that ppl might be relying on them being pulled in by asm_helper.S - your new changes take care of this